subprojects/packagefiles/libexif/meson.build

```diff
@@ -2,89 +2,72 @@ project('exif', 'c', license: 'LGPL-2.1', version: '0.6.24')
-  'memory.h',
-  'stdint.h',
-  'stdlib.h',
-  'strings.h',
-  'string.h',
-  'sys/stat.h',
-  'sys/types.h',
-  'unistd.h',
-]
+cdata = configuration_data()
+cdata.set('HAVE_LOCALTIME_R', cc.has_function('localtime_r'))
+cdata.set('HAVE_LOCALTIME_S', cc.has_function('localtime_s'))
+cdata.set('ENABLE_NLS', intl_dep.found())
+cdata.set_quoted('GETTEXT_PACKAGE', '@0@-@1@'.format(meson.project_name(), meson.project_version()))
+cdata.set_quoted('LOCALEDIR', get_option('prefix') / get_option('localedir'))
 
-foreach h : check_headers
-  if cc.has_header(h)
-    cdata.set('HAVE_' + h.underscorify().to_upper(), 1)
-  endif
-endforeach
+configure_file(
+  output: 'config.h',
+  configuration: cdata,
+)
 
-configure_file(input : 'meson_config.h.meson',
-  output : 'config.h',
-  configuration : cdata)
+sources = files(
+  'libexif/apple/exif-mnote-data-apple.c',
+  'libexif/apple/mnote-apple-entry.c',
+  'libexif/apple/mnote-apple-tag.c',
+  'libexif/canon/exif-mnote-data-canon.c',
+  'libexif/canon/mnote-canon-entry.c',
+  'libexif/canon/mnote-canon-tag.c',
+  'libexif/exif-byte-order.c',
+  'libexif/exif-content.c',
+  'libexif/exif-data.c',
+  'libexif/exif-entry.c',
+  'libexif/exif-format.c',
+  'libexif/exif-gps-ifd.c',
+  'libexif/exif-ifd.c',
+  'libexif/exif-loader.c',
+  'libexif/exif-log.c',
+  'libexif/exif-mem.c',
+  'libexif/exif-mnote-data.c',
+  'libexif/exif-tag.c',
+  'libexif/exif-utils.c',
+  'libexif/fuji/exif-mnote-data-fuji.c',
+  'libexif/fuji/mnote-fuji-entry.c',
+  'libexif/fuji/mnote-fuji-tag.c',
+  'libexif/olympus/exif-mnote-data-olympus.c',
+  'libexif/olympus/mnote-olympus-entry.c',
+  'libexif/olympus/mnote-olympus-tag.c',
+  'libexif/pentax/exif-mnote-data-pentax.c',
+  'libexif/pentax/mnote-pentax-entry.c',
+  'libexif/pentax/mnote-pentax-tag.c',
+)
 
-sources = [
-    'libexif/exif-byte-order.c',
-    'libexif/exif-content.c',
-    'libexif/exif-data.c',
-    'libexif/exif-entry.c',
-    'libexif/exif-format.c',
-    'libexif/exif-gps-ifd.c',
-    'libexif/exif-ifd.c',
-    'libexif/exif-loader.c',
-    'libexif/exif-log.c',
-    'libexif/exif-mem.c',
-    'libexif/exif-mnote-data.c',
-    'libexif/exif-tag.c',
-    'libexif/exif-utils.c',
-    'libexif/apple/exif-mnote-data-apple.c',
-    'libexif/apple/mnote-apple-entry.c',
-    'libexif/apple/mnote-apple-tag.c',
-    'libexif/pentax/exif-mnote-data-pentax.c',
-    'libexif/pentax/mnote-pentax-entry.c',
-    'libexif/pentax/mnote-pentax-tag.c',
-    'libexif/canon/exif-mnote-data-canon.c',
-    'libexif/canon/mnote-canon-entry.c',
-    'libexif/canon/mnote-canon-tag.c',
-    'libexif/olympus/exif-mnote-data-olympus.c',
-    'libexif/olympus/mnote-olympus-entry.c',
-    'libexif/olympus/mnote-olympus-tag.c',
-    'libexif/fuji/exif-mnote-data-fuji.c',
-    'libexif/fuji/mnote-fuji-entry.c',
-    'libexif/fuji/mnote-fuji-tag.c'
-]
+defs = custom_target(
+  input: 'libexif/libexif.sym',
+  output: 'libexif.def',
+  command: ['def.py', '@INPUT@', '@OUTPUT@'],
+)
```

Comment:
This should really be used on all platforms, not just Windows... bit annoying to create it and then test if it's valid as a `-Wl,.....` flag everywhere though.

Reply:
Annoying indeed.